walkthroughs/1-ionic-showcase/walkthrough.adoc

```diff
@@ -0,0 +1,345 @@
+****
+
+[type=walkthroughResource,serviceName=codeready]
+.CodeReady Workspaces
+****
+* link:https://developers.redhat.com/products/codeready-workspaces/overview/[Overview, window="_blank"]
+* link:https://access.redhat.com/documentation/en-us/red_hat_codeready_workspaces_for_openshift/1.0.0/[Documentation, window="_blank"]
+****
+
+[type=walkthroughResource,serviceName=openshift]
+.Red Hat OpenShift
+****
+* link:{openshift-host}/console[Console, window="_blank"]
+* link:https://help.openshift.com/[Openshift Online Help Center, window="_blank"]
+* link:https://blog.openshift.com/[Openshift Blog, window="_blank"]
+****
+
+
+// NOTES: Till all services will be avaiable as operators inside openshift cluster we will require additional installation steps
+// Developers need to provision all services using Tech Preview. 
+// To do that please follow README in https://github.com/aerogear/mobile-services-installer
+
+[time=20]
+== Deploying the Showcase Data Sync server
+
+The Data Sync framework is a set of libaries (base on the popular Apollo GraphQL server) that allows you to easily create mobile apps that can sync data to the backend with the following features:
+
+* Offline Support
+* Conflict Resolution
+* Easy integration with other mobile services
+
+This section describes how to deploy the example DataSync server in a {integreatly-name} cluster from the OpenShift Service Catalog.
+
+. Open the link:{openshift-host}[Openshift Console].
+// TODO: DataSync templates were removed from integrately. 
+// It is not clear for the moment how they will be loaded into Walkthorugh
+// We are adding it here manually
+. Log into OpenShift from the command line:
+.. Click on your username in top right of the OpenShift console and select *Copy Login Command*.
+.. Paste the clipboard into your terminal.
+
+. Add the Data Sync template by running:
++
+----
+oc create -f https://raw.githubusercontent.com/aerogear/datasync-deployment/master/openshift/datasync-showcase.yml
+----
+
+. Create a new project with the following name:
++
+----
+data-sync-walkthrough
+----
+. In the new project provision  'Data Sync Showcase' from the service catalog.
+[NOTE]
+====
+* The form is already prefilled with all of the necessary values.
+* The only field you might want to change is
+`+AMQ Messaging User Password+`.
+** The default value is `+Password1+` in base64 encoding
+** The value _must_ be base64 encoded
+** A custom value can be created in the terminal using
+`+$ echo <password> | base64+`
+* When the create button is clicked, a warning message may be displayed
+====
++
+The hostname for the AMQ Online Broker, (needed by demo sever) is only made available after the resources from the the template have been provisioned. One more step is needed to update the ionic-showcase-server deployment with the correct
+`+MQTT_HOST+` environment variable.
+
+// TODO can this be an script it will run from CodeReady workspaces?
```

Comment:
ping @darahayes WDYT?

Reply:
I don't know much about codeready workspaces but I don't see why not. Either way it shouldn't be a massive deal because it's such a simple step.

Reply:
This is not code ready specific.
I have tried that and it will be not possible to get that until we do some kind of binding for AMQ broker. MDC currently works only with the mobile services but the use case is the same. I want my server or client to be connected (be aware of) service running on some specific URL. There are couple upstream solutions (config operators) that can do it but this might invalidate MDC itself. Leaving this for the moment.